Add maven-semantic-release

landonreed opened the pull request:
Add https://github.com/conveyal/maven-semantic-release so that each merge to master will produce a new release.

landonreed committed:
ci(semantic-release): add maven-semantic-release to travis after_success stage

evansiroky reviewed (changes requested):
This is not enough to get it to work since maven-semantic-release expects more things to be setup to release to maven-central.  We'd need to add a `maven-settings.xml` file and also all the gpg keys.  If we don't want to release to maven-central, maven-semantic-release might actually not be the right tool for this job.

evansiroky commented:
[maven-semantic-release v4.2.0](https://www.npmjs.com/package/@conveyal/maven-semantic-release/v/4.2.0) now allows releasing maven projects without deploying to maven central.

landonreed committed:
ci(semantic-release): add --skip-maven-deploy flag

Added flag to maven-semantic-release so that the step that deploys to Maven Central is skipped during a release.

landonreed commented:
:tada: This PR is included in version 3.3.0 :tada:

The release is available on [GitHub release](https://github.com/conveyal/datatools-server/releases/tag/v3.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
